src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/IOperationClassWriter.cs

```diff
@@ -22,18 +23,22 @@ internal sealed class IOperationClassWriter
         private TextWriter _writer = null!;
         private readonly string _location;
         private readonly Tree _tree;
+        private readonly Dictionary<string, AbstractNode?> _typeMap;
 
         private IOperationClassWriter(Tree tree, string location)
         {
             _tree = tree;
             _location = location;
+            _typeMap = _tree.Types.OfType<AbstractNode>().ToDictionary(t => t.Name, t => (AbstractNode?)t);
```

Comment:
casting to `(AbstractNode?)`?  It's always goign to be non-null... so this looks like it just makes _typeMap less accurate and harder to work with, right?